src/main/scala/com/exasol/common/PropertiesParser.scala

```diff
@@ -0,0 +1,57 @@
+package com.exasol.common
+
+import scala.collection.SortedMap
+
+/**
+ * A class that reads, serializes and deserializes key value properties to a string.
+ *
+ * For example, given {@code "key1=A;key2=B"} parses to a map of {@code Map('key1' -> 'A', 'key2' -> 'B')}.
+ *
+ * @param propertySeparator a separator for properties
+ * @param keyValueAssignment an assignment for key value pairs
+ */
+final case class PropertiesParser(private val propertySeparator: String, private val keyValueAssignment: String) {
+
+  /**
+   * Parses properly separated input string into key-value properties map.
+   *
+   * @return a map of key value pairs
+   */
+  def mapFromString(string: String): Map[String, String] = {
+    if (!string.contains(propertySeparator)) {
+      throw new IllegalArgumentException(
+        s"Properties input string is not separated by '$propertySeparator'."
+      )
+    }
```

Comment:
Would it produce a false negative if only one property is passed? Or no property?

Reply:
Good find, I think this should not be here, also single property should work.